### PR TITLE
feat: support toggling tag filter

### PR DESCRIPTION
**Pull request by orhun**
see the comments in #455

Now you can press `Ctrl-t` to toggle the tag filter.

Let me know if you like this and I think we can add the key binding text to the bottom.

### CODE REVIEW

**Review by AmmarAbouZor — CHANGES_REQUESTED** [review]
Very thanks for the PR :heart:
This works in the most cases but there are some edge cases that we need to consider as well, besides some app design choices that I would prefer since I'm more familiar with the ins and out of the app.

I've mentioned all the points in the comments, and if you prefer, I can take it from here and do the needed changes since they could be too complicated for someone who isn't familiar enough with the code base. 
Please let me know if you want me to continue on your work, or ask me for more info if you have any questions while developing the requested changes :+1:

File: src/app/ui/commands/mod.rs [hunk 1]
```diff
@@ -53,6 +53,7 @@ pub enum UICommand {
     MulSelExportEntries,
     ShowFilter,
     ResetFilter,
+    ToggleTagFilter,
```
[the remark this answers is not in the document]
Comment (orhun) [reply src/app/ui/commands/mod.rs]: 37719a09912dbee30774db39ce1728fc2a24507b

File: src/app/ui/commands/mod.rs [hunk 2]
```diff
@@ -174,6 +175,10 @@ impl UICommand {
                 "Reset filter",
                 "Reset the applied filter on journals",
             ),
+            UICommand::ToggleTagFilter => CommandInfo::new(
+                "Toggle Tag Filter",
```
[the remark this answers is not in the document]
Comment (orhun) [reply src/app/ui/commands/mod.rs]: 8400c0c3f03a6c072a569054298b1aeaec8777e0

File: src/app/ui/commands/entries_list_cmd.rs [hunk 3]
```diff
@@ -391,6 +396,32 @@ pub fn exec_reset_filter<D: DataProvider>(app: &mut App<D>) -> CmdResult {
...
+            .and_then(|index| tags.get(index + 1))
+            .map(|tag| FilterCriterion::Tag(tag.to_string()))
+    });
+
+    app.apply_filter(Some(Filter {
+        criteria: vec![tag_filter.or(first_tag_filter).unwrap()],
+        ..Default::default()
+    }));
+
+    Ok(HandleInputReturnType::Handled)
```
[the remark this answers is not in the document]
Comment (orhun) [reply src/app/ui/commands/entries_list_cmd.rs]: I did my best to address these points, but 3rd and 5th wasn't clear to me.

98ed25528dd1957eee0aad418b8ff9ec514448eb

**Comment by orhun** [comment]
Thanks for the review and sorry for the delay! I addressed the comments to the best of my abilities, feel free to take it from here :)

**Comment by AmmarAbouZor** [comment]
Thanks for the Fixes, I'll take it from here to fix the issues with exiting filters & Adding unit-tests and The case if the current journal has unsaved changes

**Comment by orhun** [comment]
Perfect, thank you :)

**Comment by AmmarAbouZor** [comment]
Closed in favor of #468

**Pull request closed without merging**